languages/js/src/Query.ts

```diff
@@ -113,78 +113,82 @@ export class Query {
-          if (this.#host.hasInstance(instanceId))
-            throw new DuplicateInstanceRegistrationError(instanceId);
-          this.#host.makeInstance(tag, fields, instanceId);
-          break;
-        }
-        case QueryEventKind.ExternalCall: {
-          const {
-            attribute,
-            callId,
-            instance,
-            args,
-          } = event.data as ExternalCall;
-          this.handleCall(attribute, callId, instance, args);
-          break;
-        }
-        case QueryEventKind.ExternalIsSubspecializer: {
-          const {
-            instanceId,
-            leftTag,
-            rightTag,
-            callId,
-          } = event.data as ExternalIsSubspecializer;
-          const answer = this.#host.isSubspecializer(
-            instanceId,
-            leftTag,
-            rightTag
-          );
-          this.questionResult(answer, callId);
-          break;
-        }
-        case QueryEventKind.ExternalIsa: {
-          const { instance, tag, callId } = event.data as ExternalIsa;
-          const answer = this.#host.isa(instance, tag);
-          this.questionResult(answer, callId);
-          break;
-        }
-        case QueryEventKind.ExternalUnify: {
-          const { leftId, rightId, callId } = event.data as ExternalUnify;
-          const answer = this.#host.unify(leftId, rightId);
-          this.questionResult(answer, callId);
-          break;
+          case QueryEventKind.ExternalCall: {
+            const {
+              attribute,
+              callId,
+              instance,
+              args,
+            } = event.data as ExternalCall;
+            this.handleCall(attribute, callId, instance, args);
+            break;
+          }
+          case QueryEventKind.ExternalIsSubspecializer: {
+            const {
+              instanceId,
+              leftTag,
+              rightTag,
+              callId,
+            } = event.data as ExternalIsSubspecializer;
+            const answer = this.#host.isSubspecializer(
+              instanceId,
+              leftTag,
+              rightTag
+            );
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.ExternalIsa: {
+            const { instance, tag, callId } = event.data as ExternalIsa;
+            const answer = this.#host.isa(instance, tag);
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.ExternalUnify: {
+            const { leftId, rightId, callId } = event.data as ExternalUnify;
+            const answer = this.#host.unify(leftId, rightId);
+            this.questionResult(answer, callId);
+            break;
+          }
+          case QueryEventKind.Debug:
+            const { message } = event.data as Debug;
+            if (message) console.log(message);
+            createInterface({
+              input: process.stdin,
```

Comment:
This doesn't work for me with nestjs. I think it captures the stdin/stdout to do some fancy displaying.

Reply:
Yeah I haven't done much evaluation of how well the debug stuff is actually working. I noted that as a follow-up on the original PR (https://github.com/osohq/oso/pull/320#issuecomment-674307468) and still have it as an outstanding TODO.